python/hail/dataset.py

```diff
@@ -1071,6 +1071,51 @@ def sample_variants(self, fraction, seed=1):
     @handle_py4j
     @requireTGenotype
     @typecheck_method(output=strlike,
+                      nbits_per_prob=integral)
+    def export_bgen(self, output, nbits_per_prob=8):
+        """Export variant dataset as v1.2 BGEN and SAMPLE file.
+
+        .. include:: requireTGenotype.rst
+
+        **Examples**
+
+        Import dosage data, filter variants based on INFO score, and export data to a BGEN and SAMPLE file:
+
+        >>> vds3 = hc.import_bgen("data/example3.bgen", sample_file="data/example3.sample")
+
+        >>> (vds3.filter_variants_expr("gs.infoScore().score >= 0.9")
+        ...      .export_bgen("output/infoscore_filtered"))
+
+        **Notes**
+
+        Writes out the dataset to a BGEN and SAMPLE fileset in the `Oxford spec <http://www.stats.ox.ac.uk/%7Emarchini/software/gwas/file_format.html>`__.
+        The BGEN file is `v1.2 <http://www.well.ox.ac.uk/~gav/bgen_format/bgen_format.html>`__ and the genotype probabilities are unphased and diploid.
+        The maximum rounding error is given by :math:`\\frac{1}{2^{B} - 1}` where :math:`B` is specified by the parameter ``nbits_per_prob``. For example, if :math:`B = 8`, the maximum rounding error is 1/255 or 0.0039.
+        Multiallelic variants are supported.
+
+        The information describing each variant is as follows:
+
+        - Chromosome (``v.contig``)
+        - Variant ID (``va.varid`` if defined, else Chromosome:Position:Ref:Alt)
+        - rsID (``va.rsid`` if defined, else ".")
+        - position (``v.start``)
+        - reference allele (``v.ref``)
+        - alternate alleles (``v.altAlleles``)
+
+        The sample file has 3 columns:
+
+        - ID_1 and ID_2 are identical and set to the sample ID (``s``).
+        - The third column ("missing") is set to NA for all samples.
```

Comment:
I double checked the spec. The "missing" column is the missingRate (1 - callRate). NA is the correct value to put if undefined. http://www.stats.ox.ac.uk/%7Emarchini/software/gwas/file_format.html